src/main/java/com/demonwav/mcdev/insight/ListenerEventAnnotator.java

```diff
@@ -68,6 +60,38 @@ public void annotate(@NotNull PsiElement element, @NotNull AnnotationHolder hold
             return;
         }
 
+        PsiCodeBlock methodCodeBlock = method.getBody();
+        if (methodCodeBlock == null) {
+            return;
+        }
+        PsiStatement[] methodBodyStatements = methodCodeBlock.getStatements();
+        for (PsiStatement statement : methodBodyStatements) {
+
+            PsiAnnotation eventHandler = null;
+
+            for (AbstractModuleType<?> moduleType : moduleTypes) {
+                final List<String> listenerAnnotations = moduleType.getListenerAnnotations();
+                for (String listenerAnnotation : listenerAnnotations) {
+                    PsiAnnotation annotation = modifierList.findAnnotation(listenerAnnotation);
+                    if (annotation != null) {
+                        eventHandler = annotation;
+                    }
+                }
+            }
+
+            if (statement.getText().contains("isCancelled()")) {
+                PsiAnnotationMemberValue ignoreCancelled = eventHandler.findAttributeValue("ignoreCancelled");
```

Comment:
In @gabizou's example, you would pass the `PsiMethod` to the `AbstractModuleType`.